evmlib/src/contract/payment_vault/mod.rs

```diff
@@ -0,0 +1,58 @@
+use crate::common::{Address, Amount, QuoteHash};
+use crate::contract::payment_vault::handler::PaymentVaultHandler;
+use crate::contract::payment_vault::interface::PaymentVerification;
+use crate::quoting_metrics::QuotingMetrics;
+use crate::utils::http_provider;
+use crate::Network;
+
+pub mod error;
+pub mod handler;
+pub mod implementation;
+pub mod interface;
+
+pub const MAX_TRANSFERS_PER_TRANSACTION: usize = 256;
+
+/// Helper function to return a quote for the given quoting metrics
+pub async fn get_market_price(
+    network: &Network,
+    quoting_metrics: QuotingMetrics,
+) -> Result<Amount, error::Error> {
+    let provider = http_provider(network.rpc_url().clone());
+    let payment_vault = PaymentVaultHandler::new(*network.data_payments_address(), provider);
+    payment_vault.get_quote(quoting_metrics).await
+}
+
+/// Helper function to verify whether a data payment is valid
```

Comment:
better put a comment to explain the return value is `the amount just paid to self` or `the amount paid to all payees`